packages/fgtclb/academic-projects/Classes/Upgrades/FlexFormUpgradeWizard.php

```diff
@@ -0,0 +1,156 @@
+                continue;
+            }
+
+            $flexFormData = GeneralUtility::xml2array($record['pi_flexform']);
+
+            if (isset($flexFormData ['data']['sDEF']['lDEF']['settings.hideCompletedProjects']['vDEF'])) {
+                // Migrate checkbox toggle to select option
+                if ($flexFormData['data']['sDEF']['lDEF']['settings.hideCompletedProjects']['vDEF'] === '1') {
+                    $flexFormData['data']['sDEF']['lDEF']['settings.activeState']['vDEF'] = 'active';
+                } else {
+                    $flexFormData['data']['sDEF']['lDEF']['settings.activeState']['vDEF'] = 'all';
+                }
+                unset($flexFormData ['data']['sDEF']['lDEF']['settings.hideCompletedProjects']);
+
+                // Rename settings
+                foreach (self::MIGRATE_PLUGIN_SETTINGS as $oldName => $newName) {
+                    if (isset($flexFormData['data']['sDEF']['lDEF'][$oldName]['vDEF'])) {
+                        $flexFormData['data']['sDEF']['lDEF'][$newName]['vDEF'] = $flexFormData['data']['sDEF']['lDEF'][$oldName]['vDEF'];
+                        unset($flexFormData['data']['sDEF']['lDEF'][$oldName]);
+                    }
+                }
+
+                $this->updateContentElement(
+                    $record['uid'],
+                    $this->array2xml($flexFormData)
+                );
+            }
+        }
+
+        return false;
+    }
+
+    public function updateNecessary(): bool
+    {
+        return count($this->getMigrationRecords()) > 0;
+    }
+
+    /**
+     * @return array<int, array<string, mixed>>
+     */
+    protected function getMigrationRecords(): array
+    {
+        // @todo Check if the order of upgrade wizards can be defined to avoid check for unmigrated list_type plugins
+        $queryBuilder = $this->connectionPool->getQueryBuilderForTable('tt_content');
+        $queryBuilder->getRestrictions()->removeAll();
+        return $queryBuilder
+            ->select('*')
+            ->from('tt_content')
+            ->orWhere(
+                $queryBuilder->expr()->in(
+                    'CType',
+                    $queryBuilder->quoteArrayBasedValueListToStringList(self::MIGRATE_CONTENT_TYPES_LIST)
+                ),
+                $queryBuilder->expr()->in(
+                    'list_type',
+                    $queryBuilder->quoteArrayBasedValueListToStringList(self::MIGRATE_CONTENT_TYPES_LIST)
+                ),
+            )
+            ->executeQuery()
+            ->fetchAllAssociative();
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getPrerequisites(): array
+    {
+        return [
+            DatabaseUpdatedPrerequisite::class,
+        ];
+    }
+
+    protected function updateContentElement(int $uid, string $flexform): void
+    {
+        $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tt_content');
```

Comment:
`ConnectionPool` is injected in the construcor, why is `GeneralUtility` used here instead of using the property instance ?